kyo-direct/shared/src/main/scala/kyo/Trees.scala

```diff
@@ -3,6 +3,33 @@ package kyo
 import scala.quoted.*
 
 object Trees:
+
+    class Step(private var _gotoTrees: Seq[Any]):
+        def this() = this(Seq.empty)
+
+        protected[Trees] def addTree(using q: Quotes)(tree: q.reflect.Tree): Unit =
+            _gotoTrees = _gotoTrees :+ tree
+
+        protected[Trees] def gotoTrees(using q: Quotes): Seq[q.reflect.Tree] = _gotoTrees.asInstanceOf
```

Comment:
it's painful to type it correctly, if someone has an idea on the best way to do that, without compromising on usability, that could help!